x-pack/legacy/plugins/ml/public/jobs/new_job_new/recognize/components/job_settings_form.tsx

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiAccordion,
+  EuiButton,
+  EuiDescribedFormGroup,
+  EuiFieldText,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTextAlign,
+} from '@elastic/eui';
+import { JobGroupsInput } from '../../pages/components/job_details_step/components/groups/job_groups_input';
+import { TimeRangePicker } from '../../pages/components/time_range_step/time_range_picker';
+import { ModuleJobUI, SAVE_STATE } from '../page';
+import { getTimeFilterRange } from '../../../../components/full_time_range_selector';
+import { useKibanaContext } from '../../../../contexts/kibana';
+import {
+  composeValidators,
+  maxLengthValidator,
+  patternValidator,
+} from '../../../../../common/util/validators';
+import { JOB_ID_MAX_LENGTH } from '../../../../../common/constants/validation';
+import { isJobIdValid } from '../../../../../common/util/job_utils';
+import { TimeRange } from '../../pages/components/time_range_step/time_range';
+import { usePartialState } from '../../../../components/custom_hooks';
+
+export interface JobSettingsFormValues {
+  jobPrefix: string;
+  startDatafeedAfterSave: boolean;
+  useFullIndexData: boolean;
+  timeRange: TimeRange;
+  useDedicatedIndex: boolean;
+  jobGroups: string[];
+}
+
+interface JobSettingsFormProps {
+  saveState: SAVE_STATE;
+  onSubmit: (values: JobSettingsFormValues) => any;
+  onChange: (values: JobSettingsFormValues) => any;
+  jobGroups: string[];
+  existingGroupIds: string[];
+  jobs: ModuleJobUI[];
+}
+
+export const JobSettingsForm: FC<JobSettingsFormProps> = ({
+  onSubmit,
+  onChange,
+  saveState,
+  existingGroupIds,
+  jobs,
+}) => {
+  const { from, to } = getTimeFilterRange();
+  const { currentIndexPattern: indexPattern } = useKibanaContext();
+
+  const jobPrefixValidator = composeValidators(
+    patternValidator(/^([a-z0-9]+[a-z0-9\-_]*)?$/),
+    maxLengthValidator(JOB_ID_MAX_LENGTH - Math.max(...jobs.map(({ id }) => id.length)))
+  );
+  const groupValidator = composeValidators(
+    (value: string) => (isJobIdValid(value) ? null : { pattern: true }),
+    maxLengthValidator(JOB_ID_MAX_LENGTH)
+  );
+
+  const [formState, setFormState] = usePartialState({
+    jobPrefix: '',
+    startDatafeedAfterSave: true,
+    useFullIndexData: true,
+    timeRange: {
+      start: from,
+      end: to,
+    },
+    useDedicatedIndex: false,
+    jobGroups: [] as string[],
+  });
+  const [validationResult, setValidationResult] = useState<any>({});
```

Comment:
Any plan to replace this `any`? Could this be a `Dictionary` or `Record`?

Reply:
`<Record<string, any>`